diff --git a/lib/librender.c b/lib/librender.c
index fbe487e..e7c7938 100644
--- a/lib/librender.c
+++ b/lib/librender.c
@@ -8,6 +8,7 @@
 #include "constants.h"
 #include "libvol2bird.h"
 #include "librender.h"
+#include <string.h>
 #ifdef MISTNET
 #include "../libmistnet/libmistnet.h"
 #endif
@@ -53,6 +54,19 @@ PolarVolume_t* PolarVolume_selectScansByScanUse(PolarVolume_t* volume, vol2birdS
 
 PolarScan_t* PolarVolume_getScanClosestToElevation_vol2bird(PolarVolume_t* volume, double elev);
 
+#ifndef MIN
+#define MIN(x,y) (((x) < (y)) ? (x) : (y))
+#endif
+
+#ifndef MAX
+#define MAX(x,y) (((x) < (y)) ? (y) : (x))
+#endif
+
+#ifndef ABS
+#define ABS(x) (((x) < 0) ? (-(x)) : (x))
+#endif
+
+
 /**
  * FUNCTION BODIES
  **/
diff --git a/lib/librsl.c b/lib/librsl.c
index 939800f..cd4530d 100644
--- a/lib/librsl.c
+++ b/lib/librsl.c
@@ -13,6 +13,7 @@
 #include "constants.h"
 #include "libvol2bird.h"
 #include "rave_debug.h"
+#include <string.h>
 
 // non-public function prototypes (local to this file/translation unit)
 
@@ -26,6 +27,18 @@ PolarScanParam_t* PolarScanParam_RSL2Rave(Radar *radar, float elev, int RSL_INDE
     
 int rslCopy2Rave(Sweep *rslSweep,PolarScanParam_t* scanparam);
 
+#ifndef MIN
+#define MIN(x,y) (((x) < (y)) ? (x) : (y))
+#endif
+
+#ifndef MAX
+#define MAX(x,y) (((x) < (y)) ? (y) : (x))
+#endif
+
+#ifndef ABS
+#define ABS(x) (((x) < 0) ? (-(x)) : (x))
+#endif
+
 
 // non-public function declarations (local to this file/translation unit)
 
diff --git a/src/rsl2odim.c b/src/rsl2odim.c
index fd976dd..4bbdb68 100644
--- a/src/rsl2odim.c
+++ b/src/rsl2odim.c
@@ -11,6 +11,7 @@
 #include "polarvolume.h"
 #include "libvol2bird.h"
 #include "constants.h"
+#include <string.h>
 
 void usage(char* programName, int verbose){
     fprintf(stderr,"usage: %s <RSL polar volume input> <ODIM hdf5 volume output>\n",programName);
